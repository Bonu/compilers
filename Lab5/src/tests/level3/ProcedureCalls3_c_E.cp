// MethodCallsE: Test arguments to methods (tuple types).
class Main {
  int m1(char i) {}
  int m2(int k) {}
  int m3(int p) {}
  int m4(char i, bool p) {}
  int m5(bool p, int k, char i) {}
  int m6(bool p, bool q, char i, bool r) {}


	char i;       char j;
	int a;        int b;
	bool p;    bool q;

  int x = m1(5+1);
  int y = m1(i%2*a);
  int z = m2(p, 5);
  int r = m2(i*2);
  int r1 = m2();
  int s = m3(i);
  int t1 = m3(p);
  int t = m3("cc");
  int u = m4(3, p);
  int u1 = m4(p, '3');
  int v = m4(i+a, a, '6');
  int w = m5(p, 3, a);
  int w1 = m5("cc", i%j, 5);
  int e = m6(i+j, q, a, p);
  int f = m6(i+j, q, a);
  int g = m6(i, a, i+j*a, p, '6' + '6');

  void main() {
  }
}
