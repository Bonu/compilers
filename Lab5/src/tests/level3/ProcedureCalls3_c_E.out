line 15: argument 1 in call of method `m1' has an invalid type 
          -- expects type `char' but found an expression with type `int'
line 16: argument 1 in call of method `m1' has an invalid type 
          -- expects type `char' but found an expression with type `int'
line 17: argument 1 in call of method `m2' has an invalid type 
          -- expects type `int' but found an expression with type `bool'
line 17: wrong number of arguments in call of method `m2'
line 19: wrong number of arguments in call of method `m2'
line 21: argument 1 in call of method `m3' has an invalid type 
          -- expects type `int' but found an expression with type `bool'
line 22: argument 1 in call of method `m3' has an invalid type 
          -- expects type `int' but found an expression with type `string'
line 23: argument 1 in call of method `m4' has an invalid type 
          -- expects type `char' but found an expression with type `int'
line 24: argument 1 in call of method `m4' has an invalid type 
          -- expects type `char' but found an expression with type `bool'
line 24: argument 2 in call of method `m4' has an invalid type 
          -- expects type `bool' but found an expression with type `char'
line 25: argument 1 in call of method `m4' has an invalid type 
          -- expects type `char' but found an expression with type `int'
line 25: argument 2 in call of method `m4' has an invalid type 
          -- expects type `bool' but found an expression with type `int'
line 25: wrong number of arguments in call of method `m4'
line 26: argument 3 in call of method `m5' has an invalid type 
          -- expects type `char' but found an expression with type `int'
line 27: argument 1 in call of method `m5' has an invalid type 
          -- expects type `bool' but found an expression with type `string'
line 27: argument 3 in call of method `m5' has an invalid type 
          -- expects type `char' but found an expression with type `int'
line 28: argument 1 in call of method `m6' has an invalid type 
          -- expects type `bool' but found an expression with type `int'
line 28: argument 3 in call of method `m6' has an invalid type 
          -- expects type `char' but found an expression with type `int'
line 29: argument 1 in call of method `m6' has an invalid type 
          -- expects type `bool' but found an expression with type `int'
line 29: argument 3 in call of method `m6' has an invalid type 
          -- expects type `char' but found an expression with type `int'
line 29: wrong number of arguments in call of method `m6'
line 30: argument 1 in call of method `m6' has an invalid type 
          -- expects type `bool' but found an expression with type `char'
line 30: argument 2 in call of method `m6' has an invalid type 
          -- expects type `bool' but found an expression with type `int'
line 30: argument 3 in call of method `m6' has an invalid type 
          -- expects type `char' but found an expression with type `int'
line 30: wrong number of arguments in call of method `m6'
25 errors were detected
